fitbenchmarking/fitting/controllers/ralfit_controller.py

```diff
@@ -20,42 +21,63 @@ def __init__(self, problem, use_errors):
         """
         super(RALFitController, self).__init__(problem, use_errors)
 
-        self._x = None
-        self._inform = None
+        self._popt = None
+        self._options = {}
+
 
     def setup(self):
         """
         Setup for RALFit
         """
-        pass
+        self._options["maxit"] = 500
+        if self.minimizer=="gn":
+            self._options["model"] = 1
+            self._options["nlls_method"] = 4
+        elif self.minimizer=="gn_reg":
+            self._options["model"] = 1
+            self._options["type_of_method"] = 2
+        elif self.minimizer=="hybrid":
+            self._options["model"] = 3
+            self._options["nlls_method"] = 4
+        elif self.minimizer=="hybrid_reg":
+            self._options["model"] = 3
+            self._options["type_of_method"] = 2
+        else:
+            raise RuntimeError("An undefined RALFit minmizer was selected")
+        
 
-    def _eval_f_arg_swap(self, params, x):
-        return self.problem.eval_f(x=x,
-                                   params=params,
-                                   function_id=self.function_id)
+    def _prediction_error(self, p):
+        f = self.problem.eval_f(x=self.data_x,
+                                params=p,
+                                function_id=self.function_id)
+        f = f - self.data_y
+        if self.use_errors:
+            f = f / self.data_e
 
-    def _prediction_error(self, params, x, y):
-        return self._eval_f_arg_swap(params, x) - y
+        return f
 
-    def _jac(self, params, x, y):
-        return approx_derivative(self._eval_f_arg_swap,
-                                 params,
-                                 args=tuple([x]))
+    def _jac(self, p):
+        j = approx_derivative(self._prediction_error,
+                              p)
+        return j
 
     def fit(self):
         """
         Run problem with RALFit.
         """
         self.success = False
         try:
```

Comment:
Looking at this now, I think this try, except is superfluous.

Since solve is throwing an exception, I wouldn't have thought it would be able to return something. So, unfortunately, there won't be any way to get the result.
```
self.success = False
self._popt = ral_nlls….
self.success = True
```
Would give a correct value for success.

Reply:
I'll dig around and see if I can find an example where I get an exception for too many iterations (I was seeing it as a problem more when we were solving the wrong problem with the wrong Jacobian/not weighting correctly).  

A (nearly correct) answer was being returned with a try/catch on too many iterations, but not without -- I think this might have been to do with the try/catch that wraps `fit`, though?  Again, I'll try to find an example to show this behaviour.

Reply:
Actually -- I take that back...with the code as it is, NaN is returned for both scipy and ralfit if the max number of iterations are hit -- NIST/average_difficulty has examples of both.  I therefore agree with @AndrewLister-STFC that this try/catch isn't needed -- I'll remove.